backend/typescript/services/implementations/schedulingService.ts

```diff
@@ -0,0 +1,198 @@
+import { snakeCase } from "lodash";
+import ISchedulingService from "../interfaces/schedulingService";
+import {
+  SchedulingDTO,
+  CreateSchedulingDTO,
+  UpdateSchedulingDTO,
+} from "../../types";
+import logger from "../../utilities/logger";
+import Scheduling from "../../models/scheduling.model";
+
+const Logger = logger(__filename);
+
+class SchedulingService implements ISchedulingService {
+  /* eslint-disable class-methods-use-this */
+
+  async getSchedulingById(id: number): Promise<SchedulingDTO> {
+    let scheduling: Scheduling | null;
+
+    try {
+      scheduling = await Scheduling.findByPk(Number(id));
+
+      if (!scheduling) {
+        throw new Error(`scheduling with id ${id} not found.`);
+      }
+    } catch (error) {
+      Logger.error(`Failed to get user. Reason = ${error.message}`);
+      throw error;
+    }
+
+    // TODO: retrieve volunteer ids from scheduling-volunteer
+    // table and add to returned object once volunteer table is created
+    const volunteerIds: number[] = [];
```

Comment:
Instead of returning volunteerIds, we might be returning `volunteers: [list of volunteer objects]` in the future. No need to worry about this right now, but just wanted to point it out.